packages/langium/src/grammar/validation/validator.ts

```diff
@@ -785,6 +789,108 @@ export class LangiumGrammarValidator {
+     * whether the operator should be '+=' instead. */
+    checkAssignmentOperatorMultiplicities(rule: ast.ParserRule, accept: ValidationAcceptor): void {
+        // for usual parser rules AND for fragments, but not for data type rules!
+        if (!rule.dataType) {
+            this.checkAssignmentOperatorMultiplicitiesLogic([rule.definition], accept);
+        }
+    }
+
+    private checkAssignmentOperatorMultiplicitiesLogic(startNodes: AstNode[], accept: ValidationAcceptor): void {
+        // new map to store usage information of the assignments
+        const map: Map<string, AssignmentUse> = new Map();
+
+        // top-down traversal
+        for (const node of startNodes) {
+            // TODO dürfen mehr als 1 Action auftauchen? ansonsten funktioniert das hier nicht so ganz wie gedacht!
+            this.checkNodeRegardingAssignmentNumbers(node, 1, map, accept);
+        }
+
+        // create the warnings
+        for (const entry of map.entries()) {
+            if (entry[1].counter >= 2) {
+                for (const assignment of entry[1].assignments) {
+                    if (assignment.operator !== '+=') {
+                        accept(
+                            'warning',
+                            `It seems, that you are assigning multiple values to the feature '${assignment.feature}', while you are using '${assignment.operator}' as assignment operator. Consider to use '+=' instead in order not to loose some of the assigned values.`,
+                            { node: assignment, property: 'operator' }
+                        );
+                    }
+                }
+            }
+        }
+    }
+
+    private checkNodeRegardingAssignmentNumbers(currentNode: AstNode, parentMultiplicity: number, map: Map<string, AssignmentUse>, accept: ValidationAcceptor) {
+        // the current element can occur multiple times => its assignments can occur multiple times as well
+        let currentMultiplicity = parentMultiplicity;
+        if (ast.isAbstractElement(currentNode) && isArrayCardinality(currentNode.cardinality)) {
+            currentMultiplicity *= 2; // note, that the result is not exact (but it is sufficient for the current case)!
+        }
+
+        // assignment
+        if (ast.isAssignment(currentNode)) {
+            storeAssignmentUse(map, currentNode.feature, 1 * currentMultiplicity, currentNode);
+        }
+
+        // Search for assignments in used fragments as well, since their property values are stored in the current object.
+        // But do not search in calls of regular parser rules, since parser rules create new objects.
+        if (ast.isRuleCall(currentNode) && ast.isParserRule(currentNode.rule.ref) && currentNode.rule.ref.fragment) {
+            this.checkNodeRegardingAssignmentNumbers(currentNode.rule.ref.definition, currentMultiplicity, map, accept); // TODO fragment rules are evaluated multiple times!
```

Comment:
> // TODO fragment rules are evaluated multiple times!

Why is this a problem? I see you are using a Set<Assignment|Action>. If you make it a list, this should be not a problem anymore.

Reply:
Thanks for the hint, I forgot to remove the comment

- The "problem" I thought about was a little performance improvement, but it is not worth it, since using a map for already visited fragments makes the code less readable.
- I am using the Set in order not show multiple errors for the same assignment in this case!